### PR TITLE
Respawning Tweaks

### PR DESCRIPTION
**Pull request by NanakoAC**
changes:
  - rscadd: "Respawn timers are now tracked individually for playing as animals (mice), small synthetics (drones and pAIs) and crew (everything else). This means you can now play as a mouse or drone while waiting to respawn as a full crewmember.".
  - tweak: "You can now spawn as a drone immediately upon joining as an observer, without having to wait ten minutes. There is still a cooldown between respawning as a drone if you just died as one."
  - tweak: "Slightly improved the error messages if you try to respawn when you've not waited long enough."

In addition, unlisted changes:
The admin verb 'Allow player to respawn' now sets all three respawn timers
Respawn times are now centralised in setup #defines, for easier editing in future
Nonliving mobs will no longer attempt to ghost when deleted. A comment in mob.dm explains this change

### CODE REVIEW

File: code/setup.dm [hunk 1]
```diff
@@ -1032,3 +1032,18 @@ var/list/be_special_flags = list(
 	dview_mob.see_invisible = invis_flags; \
 	for(type in view(range, dview_mob))
 #define END_FOR_DVIEW dview_mob.loc = null
+
+
+//Time of Death constants
+//Used with a list in preference datums to track times of death
+#define	CREW 	"crew"//Used for crewmembers, AI, cyborgs, nymphs, antags
```
Comment (skull132) [inline code/setup.dm]: Space here, after the key word.

File: code/modules/mob/new_player/new_player.dm [hunk 2]
```diff
@@ -107,8 +107,7 @@
 
 			if(alert(src,"Are you sure you wish to observe? You will have to wait 30 minutes before being able to respawn!","Player Setup","Yes","No") == "Yes")
 				if(!client)	return 1
-				var/mob/dead/observer/observer = new()
-
+				var/mob/dead/observer/observer = new /mob/dead/observer(src)
```
Comment (skull132) [inline code/modules/mob/new_player/new_player.dm]: For future ref, `var/mob/dead/observer/observer = new(src)` gives the same result. No need to change atm, tho.

File: code/modules/mob/death.dm [hunk 3]
```diff
@@ -80,6 +80,12 @@
 		healths.icon_state = "health6"
 
 	timeofdeath = world.time
+	if (isanimal(src))
+		set_death_time(ANIMAL, world.time)
+	else if (ispAI(src) || isdrone(src))
+		set_death_time(MINISYNTH, world.time)
+	else if (isliving(src))
```
Comment (skull132) [inline code/modules/mob/death.dm]: If crew is fallback, change this to `else` as it is in the other file as well. That way, _at least_ this one will be set, if none other is.

**Comment by skull132** [comment]
Changelog is .txt.

**Comment by skull132** [comment]
Two changes still required, first and third comment, and the changelog is broken.

File: html/changelogs/Nanako-Respawn.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+# Optional: Remove this file after generating master changelog.  Useful for PR changelogs that won't get used again.
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes: 
+  - rscadd: "Respawn timers are now tracked individually for playing as animals (mice), small synthetics (drones and pAIs) and crew (everything else). This means you can now play as a mouse or drone while waiting to respawn as a full crewmember.".
```
Comment (skull132) [inline html/changelogs/Nanako-Respawn.yml]: Found the mistake here. It ends with a full stop, whereas it should end with a quotation mark.

File: code/modules/mob/death.dm (hunk 3 above)
Comment (NanakoAC) [inline code/modules/mob/death.dm]: no thats deliberate, a nonliving mob dying is not relevant

File: code/setup.dm (hunk 1 above)
Comment (NanakoAC) [inline code/setup.dm]: i don't understand what you mean, can you clarify

**Pull request merged**